x-pack/plugins/security_solution/public/common/components/exceptions/add_exception_modal/index.tsx

```diff
@@ -0,0 +1,337 @@
+import { useFetchOrCreateRuleExceptionList } from '../use_fetch_or_create_rule_exception_list';
+import { AddExceptionComments } from '../add_exception_comments';
+import {
+  enrichExceptionItemsWithComments,
+  enrichExceptionItemsWithOS,
+  defaultEndpointExceptionItems,
+  entryHasListType,
+  entryHasNonEcsType,
+} from '../helpers';
+import { useFetchIndexPatterns } from '../../../../alerts/containers/detection_engine/rules';
+
+// TODO: move somewhere else?
+// TODO: rename?
+export interface AddExceptionOnClick {
+  ruleName: string;
+  ruleId: string;
+  ruleExceptionLists?: ExceptionListSchema[];
+  exceptionListType: 'endpoint' | 'detection';
+  alertData: TimelineNonEcsData[] | undefined;
+}
+
+interface AddExceptionModalProps {
+  ruleName: string;
+  ruleId: string;
+  exceptionListType: 'endpoint' | 'detection';
+  alertData?: TimelineNonEcsData[];
+  onCancel: () => void;
+  onConfirm: () => void;
+}
+
+const Modal = styled(EuiModal)`
+  ${({ theme }) => css`
+    width: ${theme.eui.euiBreakpoints.m};
+  `}
+`;
+
+const ModalHeader = styled(EuiModalHeader)`
+  ${({ theme }) => css`
+    flex-direction: column;
+    align-items: flex-start;
+  `}
+`;
+
+const ModalHeaderSubtitle = styled.div`
+  ${({ theme }) => css`
+    color: ${theme.eui.euiColorMediumShade};
+  `}
+`;
+
+const ModalBodySection = styled.section`
+  ${({ theme }) => css`
+    padding: ${theme.eui.euiSizeS} ${theme.eui.euiSizeL};
+
+    &.builder-section {
+      overflow-y: scroll;
+    }
+  `}
+`;
+
+export const AddExceptionModal = memo(function AddExceptionModal({
+  ruleName,
+  ruleId,
+  exceptionListType,
+  alertData,
+  onCancel,
+  onConfirm,
+}: AddExceptionModalProps) {
+  const { http } = useKibana().services;
+  const [comment, setComment] = useState('');
+  const [shouldCloseAlert, setShouldCloseAlert] = useState(false);
+  const [shouldBulkCloseAlert, setShouldBulkCloseAlert] = useState(false);
+  const [shouldDisableBulkClose, setShouldDisableBulkClose] = useState(false);
+  const [exceptionItemsToAdd, setExceptionItemsToAdd] = useState<
+    Array<ExceptionListItemSchema | CreateExceptionListItemSchema>
+  >([]);
+  const [fetchOrCreateListError, setFetchOrCreateListError] = useState(false);
+  const [, dispatchToaster] = useStateToaster();
+  const { loading: isSignalIndexLoading, signalIndexName } = useSignalIndex();
+
+  const [{ isLoading: indexPatternLoading, indexPatterns }] = useFetchIndexPatterns(
+    signalIndexName !== null ? [signalIndexName] : []
+  );
+
+  const onError = useCallback(
+    (error: Error) => {
+      errorToToaster({ title: i18n.ADD_EXCEPTION_ERROR, error, dispatchToaster });
+      onCancel();
+    },
+    [dispatchToaster, onCancel]
+  );
+  const onSuccess = useCallback(() => {
+    displaySuccessToast(i18n.ADD_EXCEPTION_SUCCESS, dispatchToaster);
+    onConfirm();
+  }, [dispatchToaster, onConfirm]);
+
+  const [{ isLoading: addExceptionIsLoading }, addOrUpdateExceptionItems] = useAddOrUpdateException(
```

Comment:
Mentioned it above, but I think you can just use the existing `persist_exception_item.ts` hook in the lists plugin.